src/simplify_algebra.cpp

```diff
@@ -1134,71 +1135,108 @@ struct find_splits
     }
 
     /**
-     * Check if we can reach ins2 from ins1 by going through inputs of ins1
+     * Check if we can reach ins2 from ins1 by going through inputs of ins1.
+     * root is the instruction before the slice instructions (what find_splits matcher matches).
+     * Uses a BFS upwards that stops at root instruction or no inputs.
      */
-    static bool is_dependent(const module& m, instruction_ref ins1, instruction_ref ins2)
+    static bool
+    is_dependent(const module& m, instruction_ref root, instruction_ref ins1, instruction_ref ins2)
```

Comment:
If `ins1` always precedes `ins2`(which should be the case), then we can use the `reaches` function I added in #3870 here: https://github.com/ROCm/AMDMIGraphX/pull/3870/files#diff-679ea22dfd0ffd35b16186f70f94d92c84f8db9fc28ecca2bb51573071944fceR551

Reply:
I don't see how `ins1` will always precede `ins2` based on how `is_dependent` is used currently.

Reply:
That'll work. Need `reaches` merged in first though.